pyvisa_py/usb.py

```diff
@@ -292,6 +295,39 @@ def list_resources() -> List[str]:
                         "usb_interface_number": intfc,
                     },
                 )
+                logging_level = logger.getEffectiveLevel()
+                if logging_level <= logging.DEBUG:
+                    if exc_strs := traceback.format_exception(err):
```

Comment:
what's the benefit of doing this over `logger.debug("error while reading serial number", exc_info=err)`?

I am also not sure if this amount of pre-processing is worth it. On `DEBUG` I'd prefer to have all exceptions logged in general. For higher levels, we can just add the exception message to the warning's dict maybe?

Reply:
I was unaware of that kwarg to logger methods! Will change to using `exc_info`.

Reply:
I agree with @arr-ee here. Please update the code accordingly.